Userland/Libraries/LibWeb/HTML/NavigatorBeacon.cpp

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2023, Bastiaan van der Plaat <bastiaan.v.d.plaat@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <LibWeb/DOM/Document.h>
+#include <LibWeb/Fetch/Fetching/Fetching.h>
+#include <LibWeb/Fetch/Infrastructure/FetchAlgorithms.h>
+#include <LibWeb/HTML/Navigator.h>
+#include <LibWeb/HTML/NavigatorBeacon.h>
+#include <LibWeb/HTML/Window.h>
+
+namespace Web::HTML {
+
+// https://w3c.github.io/beacon/#sendbeacon-method
+WebIDL::ExceptionOr<bool> NavigatorBeaconMixin::send_beacon(String const& url, Optional<Fetch::BodyInit> const& data)
+{
+    auto& navigator = reinterpret_cast<Navigator&>(*this); // FIXME: This is ugly
+    auto& realm = navigator.realm();
+    auto& vm = realm.vm();
+    auto& relevant_settings_object = navigator.global_object().document()->relevant_settings_object();
```

Comment:
this is not the right way to get this's relevant settings object. You should be calling the free function `relevant_settings_object(*this)` from HTML/Scripting/Environments.h